plugins/database/mysql/mysql.go

```diff
@@ -182,10 +183,11 @@ func (m *MySQL) CreateUser(ctx context.Context, statements dbplugin.Statements,
 
 				return "", "", err
 			}
-			defer stmt.Close()
 			if _, err := stmt.ExecContext(ctx); err != nil {
+				stmt.Close()
```

Comment:
How come this statement isn't going through the new functions?

Reply:
Because of the custom error checking above. The code got really gross if I tried to put that into the shared logic.